From 8457f3eff89dae35d43f679a66842ceedfd08808 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?IOhannes=20m=20zm=C3=B6lnig?=
 <zmoelnig@umlautQ.umlaeute.mur.at>
Date: Fri, 13 Nov 2015 22:33:20 +0100
Subject: [PATCH] fix compilation on BigEndian

cannot pass a function by reference where an lvalue is expected
(only applies to bigendian, where a macro expands to a byteswap function)

Closes https://github.com/assimp/assimp/issues/613

Taken from [1] for buildroot assimp package compile fix.

[1] https://github.com/assimp/assimp/commit/756cfd4f74b866e3183caede69daa8c105b73bab.patch

Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 code/Bitmap.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/code/Bitmap.cpp b/code/Bitmap.cpp
index 13ec372..829fd02 100644
--- a/code/Bitmap.cpp
+++ b/code/Bitmap.cpp
@@ -84,7 +84,12 @@ namespace Assimp {
 
     template<typename T>
     inline std::size_t Copy(uint8_t* data, T& field) {
+#ifdef AI_BUILD_BIG_ENDIAN
+        T field_swapped=AI_BE(field);
+        std::memcpy(data, &field_swapped, sizeof(field)); return sizeof(field);
+#else
         std::memcpy(data, &AI_BE(field), sizeof(field)); return sizeof(field);
+#endif
     }
 
     void Bitmap::WriteHeader(Header& header, IOStream* file) {
-- 
2.1.4

